metals/src/main/scala/scala/meta/metals/MetalsServices.scala

```diff
@@ -485,24 +473,19 @@ class MetalsServices(
     Cursor(Uri(td.uri), contents, offset)
   }
 
-  private def loadAllRelevantFilesInThisWorkspace(): Unit = {
-    Workspace.initialize(cwd) { path =>
-      onChangedFile(path)(_ => ())
-    }
-  }
-
   private def onChangedFile(
       path: AbsolutePath
-  )(fallback: AbsolutePath => Unit): Unit = {
+  )(fallback: => Unit): Unit = {
```

Comment:
Out of curiosity, why did you remove the callback argument? I can imagine it could be useful at some point, but we could add it then.

Reply:
Because it was used as `fallback(path)` i.e. with the `path` argument which is in the scope anyway, i.e. when you call `onChangedFile(x) { path => ... }`, that `path` will be exactly the same `x`.